public/clinics.js

```diff
@@ -113,46 +157,53 @@ function addDeleteButtonHandler(deleteButton, index) {
     });
 }
 
-const form = document.getElementById('company-form');
-form.addEventListener('input', (e) => {
-    // If the target of the event is an address input
-    if (e.target.classList.contains('address')) {
-        // Get the parent section of the input
-        const section = e.target.parentElement;
-
-        // If the input is filled in and the section is the last one
-        if (e.target.value && section === form.lastElementChild) {
-            addBlankClinicAtBottom();
-        }
-    }
-});
+newClinicButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+    addBlankClinicAtBottom();
+})
 
 submitButton.addEventListener('click', async (e) => {
     e.preventDefault();
-
-    // Get the form values, exclude empty fields
-    const docNames = Array.from(form.getElementsByClassName('docId'), label => label.textContent);;
-    const companyNames = Array.from(form.getElementsByClassName('companyName'), input => input.value.trim()).filter(value => value !== '');;
-    const addresses = Array.from(form.getElementsByClassName('address'), input => input.value.trim()).filter(value => value !== '');
-    // Create an array of company and address objects
-    const companies = companyNames.map((name, i) => ({ docId: docNames[i], name: companyNames[i], address: addresses[i] }));
-    let companiesArray = companies.map(company => ({
+    const companyList = form.getElementsByClassName('company-address-section')
+    const companies = [];
+    for(let i = 0; i < companyList.length; i++){
+        const companyNameField = form.getElementsByClassName('companyName')[i];
+        const addressField = form.getElementsByClassName('address')[i];
+        const abnField = form.getElementsByClassName('companyAbn')[i];
+        const postCodeField = form.getElementsByClassName('companyPostCode')[i];
+        const lineNumberField = form.getElementsByClassName('lineNumber')[i];
+
+        if(companyNameField.value == '' || addressField.value == '' || abnField.value == ''){
+            alert("Please fill in required fields")
```

Comment:
Did you test this? 
I filled in the details, then got an error about "please fill ..."
then it went back to the dashboard and when I went back in all data had been deleted.
I have asked you before to test thoroughly before submitting a PR. 
You have failed to do so again. This has damaged my data.